### PR TITLE
[PW-2611] Checking if order payment additional data is present before adding 'force_save' flag

### PR DESCRIPTION
**Pull request by acampos1916**
**Description**
If the order payment's additional data is NULL the redirect ends in error because there's an attempt to unserialize the value. We're adding an early check to prevent this.

**Tested scenarios**
3DS1 payments that return `recurring.recurringDetailReference` on `/payments/details` call won't end in unserialize error.

**Fixed issue**:  #739

### CODE REVIEW

**Review by cyattilakiss — DISMISSED** [review]

File: Controller/Process/Redirect.php [hunk 1]
```diff
@@ -205,10 +205,12 @@ public function execute()
                             $extensionAttributes = $this->getExtensionAttributes($order->getPayment());
                             $extensionAttributes->setVaultPaymentToken($paymentToken);
                             $orderPayment = $order->getPayment()->setExtensionAttributes($extensionAttributes);
-                            $add = $this->serializer->unserialize($orderPayment->getAdditionalData());
-                            $add['force_save'] = true;
-                            $orderPayment->setAdditionalData($this->serializer->serialize($add));
-                            $this->orderPaymentResource->save($orderPayment);
+                            if (!empty($orderPayment->getAdditionalData())) {
+                                $add = $this->serializer->unserialize($orderPayment->getAdditionalData());
```
Comment (peterojo) [inline Controller/Process/Redirect.php:209]:
```suggestion
                                $additionalData = $this->serializer->unserialize($orderPayment->getAdditionalData());
```

File: Controller/Process/Redirect.php [hunk 2]
```diff
@@ -205,10 +205,12 @@ public function execute()
...
+                                $add['force_save'] = true;
```
Comment (peterojo) [inline Controller/Process/Redirect.php:210]:
```suggestion
                                $additionalData['force_save'] = true;
```

File: Controller/Process/Redirect.php [hunk 3]
```diff
@@ -205,10 +205,12 @@ public function execute()
...
+                                $orderPayment->setAdditionalData($this->serializer->serialize($add));
```
Comment (peterojo) [inline Controller/Process/Redirect.php:211]:
```suggestion
                                $orderPayment->setAdditionalData($this->serializer->serialize($additionalData));
```

**Review by peterojo — CHANGES_REQUESTED** [review]
I think you could be a bit more explicit with these variable names :)

**Review by peterojo — APPROVED** [review]

**Review by cyattilakiss — APPROVED** [review]

**Pull request merged**